lib/Dialect/Comb/Transforms/LowerComb.cpp

```diff
@@ -0,0 +1,107 @@
+//===- LowerComb.cpp - Lower some ops in comb -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/Comb/CombPasses.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace circt;
+using namespace circt::comb;
+
+namespace circt {
+namespace comb {
+#define GEN_PASS_DEF_LOWERCOMB
+#include "circt/Dialect/Comb/Passes.h.inc"
+} // namespace comb
+} // namespace circt
+
+namespace {
+/// Lower truth tables to mux trees.
+struct TruthTableToMuxTree : public OpConversionPattern<TruthTableOp> {
+  TruthTableToMuxTree(MLIRContext *ctxt, bool assignToWire,
+                      uint64_t &symCounter)
+      : OpConversionPattern(ctxt), assignToWire(assignToWire),
+        symCounter(symCounter) {}
+
+private:
+  /// Get a mux tree for `inputs` corresponding to the given truth table. Do
+  /// this recursively by dividing the table in half for each input.
+  // NOLINTNEXTLINE(misc-no-recursion)
+  Value getMux(Location loc, OpBuilder &b, Value t, Value f,
+               ArrayRef<bool> table, Operation::operand_range inputs) const {
+    assert(!table.empty());
```

Comment:
Should there be an assert here that table.size = 2^inputs.size?

Reply:
Optional, but a good thing to have.